consul/resource_consul_config_entry_service_intentions.go

```diff
@@ -0,0 +1,548 @@
+												Optional:    true,
+												Elem:        &schema.Schema{Type: schema.TypeString},
+											},
+											"headers": {
+												Type:        schema.TypeList,
+												Description: "Specifies a header name and matching criteria for HTTP request headers.",
+												Optional:    true,
+												Elem: &schema.Resource{
+													Schema: map[string]*schema.Schema{
+														"name": {
+															Type:        schema.TypeString,
+															Required:    true,
+															Description: "Specifies the name of the header to match.",
+														},
+														"present": {
+															Type:        schema.TypeBool,
+															Default:     false,
+															Optional:    true,
+															Description: "Enables a match if the header configured in the Name field appears in the request. Consul matches on any value as long as the header key appears in the request.",
+														},
+														"exact": {
+															Type:        schema.TypeString,
+															Optional:    true,
+															Description: "Specifies a value for the header key set in the Name field. If the request header value matches the Exact value, Consul applies the permission.",
+														},
+														"prefix": {
+															Type:        schema.TypeString,
+															Optional:    true,
+															Description: "Specifies a prefix value for the header key set in the Name field.",
+														},
+														"suffix": {
+															Type:        schema.TypeString,
+															Optional:    true,
+															Description: "Specifies a suffix value for the header key set in the Name field.",
+														},
+														"regex": {
+															Type:        schema.TypeString,
+															Optional:    true,
+															Description: "Specifies a regular expression pattern as the value for the header key set in the Name field.",
+														},
+														"invert": {
+															Type:        schema.TypeBool,
+															Optional:    true,
+															Description: "Inverts the matching logic configured in the Header.",
+														},
+													},
+												},
+											},
+										},
+									},
+								},
+							},
+						},
+					},
+					"precedence": {
+						Type:        schema.TypeInt,
+						Description: "The Precedence field contains a read-only integer. Consul generates the value based on name configurations for the source and destination services.",
+						Optional:    true,
+					},
+					"type": {
+						Type:        schema.TypeString,
+						Default:     "consul",
+						Description: "Specifies the type of destination service that the configuration entry applies to.",
+						Optional:    true,
+					},
+					"description": {
+						Type:        schema.TypeString,
+						Description: "Specifies a description of the intention.",
+						Optional:    true,
+					},
+					"legacy_id": {
```

Comment:
```suggestion
					"legacy_id": {
```

We should not expose any legacy attribute through Terraform: they might be remove in the future and nobody is relying on them when using Terraform for now so let's not add them for now and if someone needs them we can always add them later.